File: 16-06-2025_Maximum_Difference_Between_Increasing_Elements.py
class Solution:
    def maximumDifference(self, nums):
        smallest = nums[0]
        ans = -1
        
        for i in range(1, len(nums)):
            if nums[i] < smallest:
                smallest = nums[i]
            if nums[i] - smallest > ans:
                ans = nums[i] - smallest
        
        return ans if ans > 0 else -1

File: 16-06-2025_Maximum_Difference_Between_Increasing_Elements.cpp
class Solution {
public:
int maximumDifference(vector<int>& nums)
{

    int n=nums.size();
    int smallest=nums[0];
    int ans=-1;

    for(int i=1;i<n;++i)  
    {  
        if(nums[i]<smallest)  
        {  
            smallest=nums[i];  
        }  
        
        if(ans<nums[i]-smallest)  
            ans=nums[i]-smallest;  
    }  
    if(ans<=0) return -1;  
    else return ans;  
}

};